From f6161640a3a9542b0a3550c2b5dd77759eba7f1a Mon Sep 17 00:00:00 2001
Message-Id: <f6161640a3a9542b0a3550c2b5dd77759eba7f1a.1421897934.git.chang-joon.lee@intel.com>
In-Reply-To: <149a6863e085e113eb3c4f75ec994895411447d6.1421897934.git.chang-joon.lee@intel.com>
References: <149a6863e085e113eb3c4f75ec994895411447d6.1421897934.git.chang-joon.lee@intel.com>
From: Namrta <namrta.salonie@intel.com>
Date: Tue, 18 Nov 2014 21:52:01 +0530
Subject: [PATCH 5/7] FOR_UPSTREAM [VPG]: drm/i915 : Fix for access after NULL
 check in postclose function.

Found by static analysis tool.

v2: Fix commit msg & indentation in code (Rafael)

v3: Remove irrelevant checks (David)

Issue: GMINL-4352
Change-Id: Iaf5944e7d68206f5431739dc7d345f11e4ebc97f
Signed-off-by: Namrta <namrta.salonie@intel.com>
---
 drivers/gpu/drm/i915/i915_dma.c |    2 --
 1 file changed, 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_dma.c b/drivers/gpu/drm/i915/i915_dma.c
index 3cb9255..e25b906 100644
--- a/drivers/gpu/drm/i915/i915_dma.c
+++ b/drivers/gpu/drm/i915/i915_dma.c
@@ -2111,8 +2111,6 @@ void i915_driver_postclose(struct drm_device *dev, struct drm_file *file)
 	struct drm_i915_file_private *file_priv = file->driver_priv;
 
 	kfree(file_priv->process_name);
-	if (file_priv && file_priv->bsd_ring)
-		file_priv->bsd_ring = NULL;
 	kfree(file_priv);
 }
 
-- 
1.7.9.5

